test/e2e_node/runtime_conformance_test.go

```diff
@@ -253,40 +253,66 @@ while true; do sleep 1; done
 						defer f.Client.Secrets(f.Namespace.Name).Delete(secret.Name)
 						container.ImagePullSecrets = []string{secret.Name}
 					}
+					const flakeRetry = 3
+					for i := 1; i <= flakeRetry; i++ {
+						var err error
+						func() {
```

Comment:
Why enclosing the test in a function? Line 257 to 313 has become a bit hard to read for me due to the multiple nested functions and loops. Consider factoring out some part as an actual function, or use existing library code for polling/retry.